stock_inventory_valuation_report/static/src/js/stock_inventory_valuation_report_backend.js

```diff
@@ -0,0 +1,116 @@
+odoo.define(
+    "stock_inventory_valuation_report.stock_inventory_valuation_report_backend",
+    function (require) {
+        "use strict";
+
+        var core = require("web.core");
+        var AbstractAction = require("web.AbstractAction");
+        var ReportWidget = require("web.Widget");
+
+        var report_backend = AbstractAction.extend({
+            hasControlPanel: true,
+            // Stores all the parameters of the action.
+            events: {
+                "click .o_stock_inventory_valuation_report_print": "print",
+                "click .o_stock_inventory_valuation_report_export": "export",
+            },
+            init: function (parent, action) {
+                this._super.apply(this, arguments);
+                this.actionManager = parent;
+                this.given_context = {};
+                this.odoo_context = action.context;
+                this.controller_url = action.context.url;
+                if (action.context.context) {
+                    this.given_context = action.context.context;
+                }
+                this.given_context.active_id =
+                    action.context.active_id || action.params.active_id;
+                this.given_context.model = action.context.active_model || false;
+                this.given_context.ttype = action.context.ttype || false;
+            },
+            willStart: function () {
+                return Promise.all([
+                    this._super.apply(this, arguments),
+                    this.get_html(),
+                ]);
+            },
+            set_html: function () {
+                var self = this;
```

Comment:
I think use `self` variable isn't correct. For using global `this` into functions you can use this function format `() => {}`.
For example:
```js
def.then(() => {
    this.report_widget.$el.html(this.html);
});
```